helm-charts/basehub/values.yaml

```diff
@@ -1447,17 +1455,32 @@ jupyterhub:
                   timeout=spawner.k8s_api_request_retry_timeout
               )
           c.Spawner.pre_spawn_hook = ensure_db_pvc
-      05-profile-groups: |
-        # Re-assignes c.KubeSpawner.profile_list to a callable that filters the
-        # initial configuration of profile_list based on the user's github
-        # org/team membership as declared via "allowed_groups" read from
-        # profile_list profiles.
+      05-auth-state-groups-key: |
+        # Pass the user's GitHub teams from auth_state as groups to the authenticator
+        async def custom_auth_state_groups_key(auth_state):
+          groups_list = []
+          if auth_state is None:
+            print("Auth state is None")
+            return groups_list
+          else:
+            for team in auth_state["teams"]:
+              if f'{team["organization"]["login"]}:{team["slug"]}' not in c.GitHubOAuthenticator.allowed_organizations:
+                continue
+              else:
+                groups_list.append(f'{team["organization"]["login"]}:{team["slug"]}')
+            return groups_list
+        if c.GitHubOAuthenticator:
```

Comment:
You can use `hasattr` here as an additional guard, so you won't need to put empty `GitHubOAuthenticator` entries in various pieces of config.

Reply:
What object attribute would I consider applying `hasattr` to?

I thought about invoking `custom_auth_state_groups_key(auth_state)` to determine whether `groups_list` is populated, and if so, using that as a condition to set `c.GitHubOAuthenticator.auth_state_groups_key`, but this requires the `auth_state` input outside of runtime so not possible.

I am missing something obvious here 😅

Reply:
I've updated this to say

```python
        if c.JupyterHub.authenticator_class == "github":
          c.GitHubOAuthenticator.auth_state_groups_key = custom_auth_state_groups_key
```

and happy with how that works for now.